src/core/layertreemodel.h

```diff
@@ -23,6 +23,73 @@ class QgsLayerTree;
 class QgsLayerTreeModel;
 class QgsProject;
 
+class FlatLayerTreeModel : public QAbstractProxyModel
+{
+    Q_OBJECT
+
+    Q_PROPERTY( QString mapTheme READ mapTheme WRITE setMapTheme NOTIFY mapThemeChanged )
+
+  public:
+    enum Roles
+    {
+      VectorLayerPointer = Qt::UserRole + 1,
```

Comment:
Could this be renamed to VectorLayer? In QML we shouldn't be talking any C++ speek like "pointer".

Reply:
We can't because we register VectorLayer as a QML object type here:
https://github.com/opengisch/QField/blob/master/src/core/qgismobileapp.cpp#L187

Reply:
Ok, let's stick to it this way then.
Fun story "layer" seems to be a reserved word in qml too (never found out why or where)